### ISSUE TITLE
UI improvements for foldable devices

### ISSUE THREAD

**Issue by kaibagley** [MM2-0/Kvaesitso#1088]
Hey everyone, 

I made a discussion: https://github.com/MM2-0/Kvaesitso/discussions/1081, but I think this is more suitable as a feature request/issue.

1. Variable apps-per-row in app drawer: When swapping between the thin front screen and wide unfolded screen, it would be great for the number of apps to respond to the width of the screen. At the moment I have 8, which looks fine for the wide screen, but is very tightly packed for the thin screen. Kvaesitso could just double the number of columns in the drawer if the device is open which might be the simplest way to deal with this.
2. Multiple columns for widgets: When scrolling down to the widgets, Kvaesitso fills the width of the screen with the widget, causing them to display as super wide. I think a good solution to this would be to have 2 columns when the device is open.

Thanks for the fantastic launcher, I wish I could contribute some code to this issue, however I know nothing about app development, and less about Kotlin....

**Comment by StormPooper** [comment]
Another potential option - left screen shows the home screen, right shows the widgets view.

**Comment by Astrrra** [comment]
+1 to this, it's so close to being absolutely perfect for me but having a single setting for how many rows of apps I have regardless of whether it's the inner/outer screen drives me crazy...

**Comment by wipe-tartly-polar** [comment]
Can I also second this request?

**Comment by shakrunk** [comment]
I'd like to take a first pass at point 1 (variable apps-per-row in the app drawer).

Proposed approach: rather than adding a separate "columns when unfolded" setting, treat the existing column count setting as the baseline for a compact-width window, and scale it by the current window's Material width size class (compact/medium/expanded, per the [M3 breakpoints](https://m3.material.io/foundations/layout/applying-layout/window-size-classes)) — doubling it at medium/expanded width. That should cover the "double the columns on the wide screen" behavior with no new settings UI. I'm computing the width class from the window size that's already piped through the compose tree, so no new library dependency is needed.

Scope for a first PR: app drawer / search result grid, favorites edit sheet, tag edit sheet, transparency preview — i.e. everywhere the shared grid-columns setting is consumed. The dock/clock-widget inline favorites row reads the column count independently (outside Compose, in a ViewModel) and isn't covered by this pass — can follow up separately.

Point 2 (multi-column widgets) is a bigger change — the widget column is currently a single vertical list with drag-to-reorder, so multi-column needs reorder-across-columns logic. I'll look at that as a follow-up once the grid part lands, to keep the PR reviewable.

Flagging here first per the contributor guide before opening a PR, to avoid duplicate work if anyone else is already on this.

**Comment by MM2-0** [comment]
Sounds good @shakrunk

**Comment by shakrunk** [comment]
I've opened #1978 implementing the first half of this: the app drawer / search grid doubles its column count when the window is ≥ 600dp wide (the M3 medium/expanded breakpoints), using the existing column setting as the compact-screen baseline.

Here it is on a Galaxy Z Fold3 with the columns setting at 6 — same install, just folded vs. unfolded:

| Folded (cover display): 6 columns | Unfolded (inner display): 12 columns |
|---|---|
| <img src="https://raw.githubusercontent.com/shakrunk/Kvaesitso/pr-1978-assets/folded-drawer.png" width="280"/> | <img src="https://raw.githubusercontent.com/shakrunk/Kvaesitso/pr-1978-assets/unfolded-drawer.png" width="450"/> |

One heads-up for your setup: since the setting now describes the *folded* screen, your 8 columns would become 16 unfolded — you'd set it to 4 to get 8 on the inner display. Feedback on whether that mapping feels right (vs. a separate setting per screen) is welcome on the PR.

The multi-column widget layout half is planned as a follow-up PR to keep the first one reviewable.
